pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/partition/ByteArrayPartitionFunction.java

```diff
@@ -40,8 +42,8 @@ public ByteArrayPartitionFunction(int numPartitions) {
   }
 
   @Override
-  public int getPartition(Object valueIn) {
-    return abs(Arrays.hashCode(valueIn.toString().getBytes())) % _numPartitions;
+  public int getPartition(Object value) {
+    return (Math.abs(Arrays.hashCode(value.toString().getBytes(UTF_8))) & Integer.MAX_VALUE) % _numPartitions;
```

Comment:
Sorry, what I meant was 

```java
Math.abs(Arrays.hashCode(value.toString().getBytes(UTF_8)) % _numPartitions);
```

produces the right value, because `Arrays.hashCode(value.toString().getBytes(UTF_8)))  % _numPartitions` takes a value in the range `(-_numPartitions, _numPartitions)` which excludes `Integer.MIN_VALUE`, - `Math.abs` is defined everywhere on this domain.

Reply:
Oh nice. Didn't realize these 2 are the same. Applied the change

Reply:
Actually they are not the same if the hash code is `Integer.MIN_VALUE`, where the current one will always result in `0`. We are trying to match the Kafka [ByteArrayPartitioner](https://github.com/a0x8o/kafka/blob/master/core/src/main/scala/kafka/producer/ByteArrayPartitioner.scala) which uses this [abs()](https://github.com/a0x8o/kafka/blob/master/clients/src/main/java/org/apache/kafka/common/utils/Utils.java#L179).

Let me add some comments on this

Reply:
Ok I didn’t realise there was a predefined output requirement, just that min value needed to end up non negative somehow. Makes sense.